lib/restbase.js

```diff
@@ -243,6 +243,10 @@ RESTBase.prototype._request = function(req) {
     var match = priv.router.route(childReq.uri);
     var methods = match && match.value && match.value.methods;
     var handler = methods && (methods[childReq.method] || methods.all);
+    if (childReq.method === 'head' && !handler) {
```

Comment:
I'm wondering if we should store the original request's method (in the Restbase object) and propagate it. Then we could decide on the right handler every time, not just until we fail for the first time. Consider a situation where the public endpoint, *PE*, calls an internal one, *S1*, which calls another one, *S2*. Now, if *S2* had both `GET` and `HEAD` handlers, but only `GET`s were registered for *S1*, *S2*'s `HEAD` handler would not be called. Instead, the `GET` one would be used, potentially entailing a wrongful execution. Hence, *S2* should be able to use the original request's method as well when deciding which handler to use.